vertx-web-client/src/main/java/io/vertx/ext/web/client/impl/HttpRequestImpl.java

```diff
@@ -84,6 +85,7 @@ private HttpRequestImpl(HttpRequestImpl<T> other) {
     this.params = other.params != null ? new CaseInsensitiveHeaders().addAll(other.params) : null;
     this.codec = other.codec;
     this.followRedirects = other.followRedirects;
+    this.hostHeader = other.hostHeader;
```

Comment:
Are we losing this parameter in the other constructor?

Reply:
This is required for copy constructor. The other constructor never initialises this value by desigm. It needs to be set through setHost() method only - analogously to HttpClient in vertx-core